handler/handler_test.go

```diff
@@ -223,6 +225,73 @@ func TestNew(t *testing.T) {
 	}
 }
 
+func TestReverseProxyFatal(t *testing.T) {
+	type args struct {
+		cfg  config.Proxy
+		bp   httputil.BufferPool
+		prov service.Authorizationd
+	}
+	type test struct {
+		name      string
+		args      args
+		checkFunc func(http.Handler) error
+	}
+	tests := []test{
+		func() test {
+			handler := http.HandlerFunc(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Write([]byte("dummyContent"))
+				w.WriteHeader(http.StatusOK)
+			}))
+			srv := httptest.NewServer(handler)
+
+			return test{
+				name: "check fatal, new request failed",
+				args: args{
+					cfg: config.Proxy{
+						Host: "invalid_URL_@@@",
+						Port: func() uint16 {
+							a, _ := strconv.ParseInt(strings.Split(srv.URL, ":")[2], 0, 64)
+							return uint16(a)
+						}(),
+					},
+					bp: infra.NewBuffer(64),
+					prov: &service.AuthorizerdMock{
+						VerifyFunc: func(r *http.Request, act, res string) error {
+							return nil
+						},
+					},
+				},
+				checkFunc: func(h http.Handler) error {
+					rw := httptest.NewRecorder()
+					r := httptest.NewRequest("GET", "http://dummy.com", nil)
+					h.ServeHTTP(rw, r)
+					if rw.Code != http.StatusOK {
+						return errors.Errorf("unexpected status code, got: %v, want: %v", rw.Code, http.StatusOK)
+					}
+					return nil
+				},
+			}
+		}(),
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Do you need a comment to specify that the test is code for not fatal stops?

It has nothing to do with the PR code, but I thought that NewRequest()'s fatal stop processing should be done at program startup, not just with `Director`.

Reply:
> Do you need a comment to specify that the test is code for not fatal stops?

The func name is `TestReverseProxyFatal`. What comment should be added?

> It has nothing to do with the PR code

yes, it is to test the old non-covered code.
https://github.com/yahoojapan/authorization-proxy/blob/e0afb779813a37b29817e82b65ae8e1144725d98/handler/handler.go#L55

> I thought that NewRequest()'s fatal stop processing should be done at program startup, not just with `Director`.

Cannot get your idea, can you explain more?
The proxy can only new request when getting a request after program startup.

Reply:
created issue: https://github.com/yahoojapan/authorization-proxy/issues/51